libcudacxx/include/cuda/pipeline

```diff
@@ -514,59 +516,45 @@ _LIBCUDACXX_BEGIN_NAMESPACE_CUDA
     {
         (void)__pipeline;
         NV_IF_TARGET(NV_PROVIDES_SM_80,
-            __memcpy_arrive_on_impl::__arrive_on(__barrier, async_contract_fulfillment::async);
+            __barrier_cp_async_arrive_on(__barrier);
         )
     }
 
-    template<typename _Group, class _Type, thread_scope _Scope>
-    _LIBCUDACXX_INLINE_VISIBILITY
-    async_contract_fulfillment memcpy_async(_Group const & __group, _Type * __destination, _Type const * __source, std::size_t __size, pipeline<_Scope> & __pipeline) {
-        // When compiling with NVCC and GCC 4.8, certain user defined types that _are_ trivially copyable are
-        // incorrectly classified as not trivially copyable. Remove this assertion to allow for their usage with
-        // memcpy_async when compiling with GCC 4.8.
-        // FIXME: remove the #if once GCC 4.8 is no longer supported.
-    #if !defined(_LIBCUDACXX_COMPILER_GCC) || _GNUC_VER > 408
-        static_assert(std::is_trivially_copyable<_Type>::value, "memcpy_async requires a trivially copyable type");
-    #endif
-
-        return __memcpy_async<alignof(_Type)>(__group, reinterpret_cast<char *>(__destination), reinterpret_cast<char const *>(__source), __size, __pipeline);
-    }
+    template<thread_scope _Sco, __tx_api _Tx, typename _Arch, __space _OutSpace, __space _InSpace, __space _SyncSpace>
```

Comment:
I really dislike `class`, because it's the less semantically accurate of the two spellings (an argument could be an `int`, and `int` is definitely not a class) - but we have both throughout the library, so we should probably settle on a policy and do a library-wide unification if we want to.